client/infinite_scroll.js

```diff
@@ -1,18 +1,19 @@
-Template.infinitescroll.onCreated(function() {
-   this.showMoreVisible = function() {
-     var target = $(".showMoreResults");
-      if (!target.length) return;
-       if (document.documentElement.clientHeight + $(document).scrollTop() >= document.body.offsetHeight )
-       {
-        $(".showMoreResults").trigger("becameVisible");
-       }
-     };
-});
-
-Template.infinitescroll.onRendered(function() {
-  $(window).on('scroll',_.debounce(this.showMoreVisible, this.data.timeDelay));
-});
-
-Template.infinitescroll.onDestroyed(function() {
+TemplateController('infinite_scroll', {
+  onCreated() {
+    let instance = this;
+    this.showMoreVisible = function() {
+    instance.target = $(".showMoreResults");
+    if (!instance.target.length) return;
+    if (document.documentElement.clientHeight + $(document).scrollTop() >= document.body.offsetHeight)
+    {
+      instance.target.trigger("becameVisible");
+    }
+    };
+  },
+  onRendered() {
+    $(window).on('scroll',_.debounce(this.showMoreVisible, this.data.timeDelay));
+  },
+  onDestroyed() {
     $(window).off('scroll',_.debounce(this.showMoreVisible, this.data.timeDelay));
```

Comment:
This will not work since `_.debounce` always returns a new function, so the handler you registered in the `onRendered` function will be different from the one in `onDestroyed`. You have to pass exactly the same function reference to `on` and `off`!